spec/features/create_listings_feature_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+
+feature "Create listings" do
+  context "when not signed in" do
+    before do
+      visit "/listings"
+      click_link "Add a class"
+    end
+
+    scenario "does not allow user to add a listing" do
+      expect(page).to have_content "You need to sign in or sign up before continuing."
```

Comment:
Metrics/LineLength: Line is too long. [86/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)